app/src/main/java/com/android/voyageur/ui/components/PlaceSearchWidget.kt

```diff
@@ -20,22 +25,41 @@ import com.android.voyageur.model.place.PlacesViewModel
 fun PlaceSearchWidget(
     placesViewModel: PlacesViewModel,
     onSelect: (CustomPlace) -> Unit,
-    modifier: Modifier = Modifier
+    modifier: Modifier = Modifier,
+    query: TextFieldValue,
+    onQueryChange: (TextFieldValue) -> Unit
 ) {
-  var query by remember { mutableStateOf(TextFieldValue("")) }
   var expanded by remember { mutableStateOf(false) }
   val searchedPlaces by placesViewModel.searchedPlaces.collectAsState()
 
   Column {
     OutlinedTextField(
         value = query,
         onValueChange = {
-          placesViewModel.setQuery(it.text, null)
+          onQueryChange(it)
           expanded = it.text.isNotEmpty() && it.text != query.text
-          query = it
         },
         modifier = modifier.fillMaxWidth().testTag("searchTextField"),
-        placeholder = { Text("Search places...") })
+        keyboardOptions = KeyboardOptions.Default.copy(imeAction = ImeAction.Search),
+        keyboardActions =
+            KeyboardActions(
+                onSearch = {
+                  val matchingPlace =
+                      searchedPlaces.firstOrNull { it.place.displayName == query.text }
+                  if (matchingPlace != null) {
+                    onSelect(matchingPlace)
```

Comment:
We should remove this, we select the place only when the user clicks on an item from the dropdown, not when his query matches exactly the name of a given place